Remove docstring types where required

spyke7 opened the pull request:
## Description

**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

This PR resolves issue #126 as already sphinx-autodoc-typehints is there.

**What does this PR do?**

In this PR, the type annotations are removed from the docstrings where it is required from the python files inside `ethology`

## References

#126 

## How has this PR been tested?

Only the docstring has been changed as needed in the issue, there is no change in any type of functionality. 

## Is this a breaking change?

No this is not a breaking change.

## Does this PR require an update to the documentation?

No new feature have been added, only the docstrings are updated. I guess there is no need for docs update..

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

spyke7 committed:
Remove type annotations from docstring where required

spyke7 committed:
Remove type annotations from docstring where required

sfmig commented on ethology/io/annotations/save_bboxes.py:
```diff
@@ -26,14 +26,14 @@ def to_COCO_file(dataset: xr.Dataset, output_filepath: str | Path):
     Returns
     -------
-    str
+    - str
```
```suggestion
    str
```

sfmig reviewed (changes requested):
Hi @spyke7, thanks for this!

I commented in-line a typo in one of the docstrings (en extra hyphen), otherwise it looks good. 

However, there are some more types under the `validator` module that are not rendering as "clickable" currently in the API docs  (e.g. in the `ethology.validators.annotations` module, the `Index` type of the `idx` attribute should be [pandera.typing.Index](https://pandera.readthedocs.io/en/stable/reference/generated/pandera.typing.Index.html#pandera.typing.Index), the raised error should be `pandera.errors.SchemaError`, `ClassVar` should be `typing.ClassVar`,  etc....).

I think it would make sense to include this as part of this PR. Would you mind covering that too? Let me know!

I would recommend reading the ['Editing documentation' section](https://ethology.neuroinformatics.dev/community/contributing.html#contributing-documentation) to build and visualise the documentation locally, to check the API reference docs are rendering with all "clickable" types.

spyke7 committed:
removed extra hyphen

spyke7 committed:
Merge branch 'main' into remove_docstring_types

spyke7 commented:
ok @sfmig, I will try to solve the part of `validators` as well.

spyke7 committed:
made types clickable

spyke7 commented:
@sfmig I have updated the annotation.py, uitls.py and detections.py under validators.
Previously the import was `from typing import ClassVar`, now I have changed it `import typing`, and modified to this `typing.ClassVar[...]`
This I guess solves the thing. But let me know if I need to update anything

sfmig committed:
Revert "made types clickable"

This reverts commit c6e38d46ab23b9e487111688225737a8d55ffdf2.

sfmig committed:
Spell out module paths in docstrings of validator classes

sfmig committed:
Replace pytz

sfmig commented:
Hi @spyke7, thanks for this!

However I realised your last commit did not fix the last thing we discussed, and the types of the attributes of the validator classes were still not clickable in the API docs. I reverted the commit and implemented the changes, but I explain why below in case you are interested.

With `sphinx-autodoc-typehints`, we can rely on the regular python typing for sphinx to fetch the types of _input_ arguments to _functions_. So your first few commits removed the redundancy by deleting the types from the docstring in these cases. However, this automatic type inference does not work with _classes_ at the moment. So for these we have to spell out the path to the type module in the docstring of the class. Your commit modified the way the modules were imported in the code section of the class, but not the docstring, so it didn't change the rendering of the API and the types for the validator class attributes were still not 'clickable'. 

I understand this particular case was a bit tricky, as functions and classes are treated differently, but as I mentioned in my last comment, it is important to build the docs locally when contributing to the docs to make sure the edits are working and rendering as expected. If you check this before marking a docs PR as ready for review, you will catch many errors much earlier, and the reviewer will be faster to review and give you feedback.

spyke7 commented:
> However I realised your last commit did not fix the last thing we discussed, and the types of the attributes of the validator classes were still not clickable in the API docs. I reverted the commit and implemented the changes, but I explain why below in case you are interested.
> 
> With `sphinx-autodoc-typehints`, we can rely on the regular python typing for sphinx to fetch the types of _input_ arguments to _functions_. So your first few commits removed the redundancy by deleting the types from the docstring in these cases. However, this automatic type inference does not work with _classes_ at the moment. So for these we have to spell out the path to the type module in the docstring of the class. Your commit modified the way the modules were imported in the code section of the class, but not the docstring, so it didn't change the rendering of the API and the types for the validator class attributes were still not 'clickable'.
> 
> I understand this particular case was a bit tricky, as functions and classes are treated differently, but as I mentioned in my last comment, it is important to build the docs locally when contributing to the docs to make sure the edits are working and rendering as expected. If you check this before marking a docs PR as ready for review, you will catch many errors much earlier, and the reviewer will be faster to review and give you feedback.

I have made the docs from my side previously, but there may be some error in it. It's fine! Thanks for guiding :)
